File: climbing-stairs.cpp
class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        a=1
        b=2
        for i in range(n-1):
            c=a+b
            a=b
            b=c
        return a
        

File: climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        if(n <= 2) return n;
        int prev = 1;
        int cur = 2;
        for(int i = 1; i < n; i++){
            int nu = prev + cur;
            prev = cur;
            cur = nu;
        }return prev;
    }
}